src/resources/2015/Dinsey.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  availableChoiceOptions,
+  myAdventures,
+  myAscensions,
+  runChoice,
+  visitUrl,
+} from "kolmafia";
+import { getRemainingLiver, have } from "../..";
+import { get, getString, getNumber } from "../../property";
+import { $location, $item } from "../../template-string";
+
+const kioskUrl = "place.php?whichplace=airport_stench&action=airport3_kiosk";
+const maintUrl = "place.php?whichplace=airport_stench&action=airport3_tunnels";
+
+export const questPreferenceKioskText = new Map<string, string>([
+  ["questEStSocialJusticeII", "Racism Reduction"],
+  ["questEStSocialJusticeI", "Sexism Reduction"],
+  ["questEStFishTrash", "Waterway Debris Removal"],
+  ["questEStZippityDooDah", "Compulsory Fun"],
+  ["questEStSuperLuber", "Track Maintenance"],
+  ["questEStGiveMeFuel", "Electrical Maintenance"],
+  ["questEStNastyBears", "Bear Removal"],
+  ["questEStWorkWithFood", "Guest Sustenance Assurance"],
+]);
+
+export const questPreferenceLocation = new Map<string, Location>([
+  [
+    "questEStSocialJusticeII",
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+  ["questEStSocialJusticeI", $location`Pirates of the Garbage Barges`],
+  ["questEStFishTrash", $location`Pirates of the Garbage Barges`],
+  ["questEStZippityDooDah", $location`The Toxic Teacups`],
+  ["questEStSuperLuber", $location`Barf Mountain`],
+  ["questEStGiveMeFuel", $location`The Toxic Teacups`],
+  [
+    "questEStNastyBears",
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+  ["questEStWorkWithFood", $location`Barf Mountain`],
+]);
+
+interface questRequirement {
+  item: Item;
+  quantity: number;
+  trackingPref: string;
+  completeValue: number;
+}
+
+export const questRequirements = new Map<string, questRequirement>([
+  [
+    "questEStSocialJusticeII",
+    {
+      item: $item`none`,
+      quantity: 0,
+      trackingPref: "dinseySocialJusticeIIProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStSocialJusticeI",
+    {
+      item: $item`none`,
+      quantity: 0,
+      trackingPref: "dinseySocialJusticeIProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStFishTrash",
+    {
+      item: $item`trash net`,
+      quantity: 1,
+      trackingPref: "dinseyFilthLevel",
+      completeValue: 0,
+    },
+  ],
+  [
+    "questEStZippityDooDah",
+    {
+      item: $item`Dinsey mascot mask`,
+      quantity: 1,
+      trackingPref: "dinseyFunProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStSuperLuber",
+    {
+      item: $item`lube-shoes`,
+      quantity: 1,
+      trackingPref: "",
+      completeValue: 0,
+    },
+  ],
+  [
+    "questEStGiveMeFuel",
+    {
+      item: $item`toxic globule`,
+      quantity: 20,
+      trackingPref: "",
```

Comment:
I would make this null probably, but I can see why that would be a pain to deal with.

Reply:
I switched the interface to allow trackingPref and completeValue to be optional (which uses undefined instead of null, but the same idea)